docs/Developer Documentation/ResumeController/index.md

```diff
@@ -0,0 +1,224 @@
+
+ResumptionData interface contains responcible for keeping application information. 
+ResumeCtrlImpl is responsible for HMI state restoring. 
+
+ResumeCtrlImpl will remove application hmi_state info from resumption data after 3 ignition cycles. 
+On each shutdown ResumeCtrlImpl will increment `ign_off_count` value for each application.
+
+On App registration `ResumeCtrl::StartResumptionOnlyHMILevel` or `ResumeCtrlImpl::StartResumption` will put aplication in a queue for resumption. 
+Internal timer in ResumeCtrlImpl will restore application hmi_state in several seconds (configured by `ApplicationManagerSettings::app_resuming_timeout`)
+In case if any  other application already registered, StateController will take care about hmi_states conflicts resolving. 
+
+### Data resumption
+
+SDL restores application data if application sends appropriate `hash` in the RAI. Hash updates after each data change. 
+SDL stores resumption data either in json or in database, this option is configurable. 
+
+ResumeControllerImpl requests app data from `ResumptionData` class and provides it to `ResumptionDataProcessor`
+
+`ResumptionDataProcessor` is responsible for restoring application data and provides the result to RAI via callback
+
+|||
+Figure 2: Resumption data seauence Overview
+![Resumption data seauence](./assets/data_resumption.png)
+|||
+
+### ResumptionData
+
+`ResumptionData` class is used to represent resumption data agnostic to data storage. 
+`ResumptionData` provides app resumption data in the Smart Object representation. 
+
+|||
+Figure 2: Resumption data classes
+![Resumption data classes](./assets/resumption_data_class.png)
+|||
+
+There are 2 implementations of resumption data : 
+ * `ResumptionDataJson`
+ * `ResumptionDataDB`
+
+ `ResumptionData` does not contain active components : timers, reactions, callbacks, etc ...
+ It behaves like data a storage.
+
+### ResumptionDataProcessor
+
+`ResumptionDataProcessor` is responsible for restoring resumption data and tracking its status. 
+
+Main public function for resumptions is `ResumptionDataProcessor::Restore` : 
+```cpp
+  /**
+  * @brief Running resumption data process from saved_app to application.
+  * @param application application which will be resumed
+  * @param saved_app application specific section from backup file
+  */
+  void Restore(app_mngr::ApplicationSharedPtr application,
+               smart_objects::SmartObject& saved_app,
+               ResumeCtrl::ResumptionCallBack callback);
+```
+
+`ResumeCtrl::ResumptionCallBack callback` is a function that should be called after data resumption : 
+```cpp
+typedef std::function<void(mobile_apis::Result::eType result_code,
+                             const std::string& info)> ResumptionCallBack;
+```
+
+Some resumption data should be restored in the `Application` class itself.
+Some resumption data should be stored in plugins : ApplicationExtensions. 
+Some resumption data requires sending HMI request. 
```

Comment:
please check the markdown, it looks like 

> Some resumption data should be restored in the Application class itself. Some resumption data should be stored in plugins : ApplicationExtensions. Some resumption data requires sending HMI request.

Reply:
fixed  dac0e3c